cosmwasm/ibc-union/core/src/contract.rs

```diff
@@ -103,6 +110,39 @@ pub mod events {
     }
 }
 
+#[must_use]
+fn packet_to_attrs(packet: &Packet) -> [Attribute; 5] {
+    [
+        (
+            events::attribute::PACKET_SOURCE_CHANNEL_ID,
+            packet.source_channel_id.to_string(),
+        ),
+        (
+            events::attribute::PACKET_DESTINATION_CHANNEL_ID,
+            packet.destination_channel_id.to_string(),
+        ),
+        (events::attribute::PACKET_DATA, packet.data.to_string()),
+        (
+            events::attribute::PACKET_TIMEOUT_HEIGHT,
+            packet.timeout_height.to_string(),
+        ),
+        (
+            events::attribute::PACKET_TIMEOUT_TIMESTAMP,
+            packet.timeout_timestamp.to_string(),
+        ),
+    ]
+    .map(Into::into)
```

Comment:
one thing to note: `array::map` is [quite inefficient](https://doc.rust-lang.org/stable/std/primitive.array.html#note-on-performance-and-stack-usage), so we may want to consider returning `impl Iterator<Item = Attribute>` here instead

Reply:
Seems to be fine for small array?
> This mainly concerns large arrays, as mapping over small arrays seem to be optimized just fine. 
Also this is a constant sized array so it should be completely erased and inlined I assume

Reply:
see: https://godbolt.org/z/6hxrh4rYx

Reply:
(wasm: https://godbolt.org/z/Y1sMaxjna)